ksql-common/src/main/java/io/confluent/ksql/schema/ksql/KsqlSchema.java

```diff
@@ -82,10 +86,70 @@ public Schema getSchema() {
     return schema;
   }
 
+  /**
+   * Get all the fields in the schema.
+   *
+   * @return all the fields in the schema.
+   */
   public List<Field> fields() {
     return schema.fields();
   }
 
+  /**
+   * Get the set of field indexes for the implicit fields, if any.
+   *
+   * @return the set of indexes to the implicit fields.
+   */
+  public Set<Integer> implicitColumnIndexes() {
+    return IMPLICIT_FIELD_NAMES.stream()
+        .map(schema::field)
+        .filter(Objects::nonNull)
+        .map(Field::index)
+        .collect(Collectors.toSet());
+  }
+
+  /**
+   * Search for a field with the supplied {@code fieldName}.
+   *
+   * <p>If the fieldName and the name of a field are an exact match, it will return that field.
+   *
+   * <p>If not exact match is found, any alias is stripped from the supplied  {@code fieldName}
+   * before attempting to find a match again.
+   *
+   * @param fieldName the field name, where any alias is ignored.
+   */
+  public Optional<Field> findField(final String fieldName) {
+    return schema.fields()
+        .stream()
+        .filter(f -> SchemaUtil.matchFieldName(f, fieldName))
+        .findFirst();
+  }
+
+  /**
+   * Find the index of the field with the supplied exact {@code fieldName}.
+   *
+   * @param fieldName the exact name of the field to get the index of.
+   * @return the index if it exists or else {@code empty()}.
+   */
+  public OptionalInt fieldIndex(final String fieldName) {
+    final Field field = schema.field(fieldName);
+    if (field == null) {
```

Comment:
nit: `return field == null ? OptionalInt.empty() : OptionalInt.of(field.index());`